### PR TITLE
CLOUDSTACK-10382: Fix cloud-get-vm-data-configdrive.in

### PR DESCRIPTION
**Pull request by fmaximus**
## Description

Fix file locations and lookup.
Cleanup of code - requires bash v4.

## Types of changes

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

## Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [ ] I have added tests to cover my changes.
- [ ] All relevant new and existing integration tests have passed.
- [ ] A full integration testsuite with all test that can run on my environment has passed.

### CODE REVIEW

File: setup/bindir/cloud-get-vm-data-configdrive.in [hunk 1]
```diff
@@ -21,23 +21,28 @@ mountdir=$(mktemp -d)
 filepath=$mountdir/cloudstack
 
 user_data=$filepath/userdata/user_data.txt
-availability_zone=$filepath/metadata/availability_zone.txt
```
Comment (rhtyd) [inline setup/bindir/cloud-get-vm-data-configdrive.in]: This assumes that bash is available, will the syntactic changes work with `sh`?

Comment (fmaximus) [reply setup/bindir/cloud-get-vm-data-configdrive.in]: The original script was only working under bash as well. function is a keyword unknown in `sh`.
As specified in the description, the changes require Bash 4.
I can also add a version specifically for bash 3, but I suppose with shellshock most people will have upgraded to bash 4 by now.

**Comment by DaanHoogland** [comment]
@PaulAngus I am fine with this as well. @rhtyd this is only a problem for client instances that use older versions of bash, isn't it?

**Comment by rhtyd** [comment]
I've not tested, but I'm okay with the change as well.

**Comment by rhtyd** [comment]
LGTM, tested the script on a config drive + L2:
screenshot from 2018-06-07 11-17-40

**Pull request merged**